apps/mobile-wallet/src/components/NFTImage.tsx

```diff
@@ -12,23 +13,51 @@ import { BORDER_RADIUS_SMALL } from '~/style/globalStyle'
 export interface NFTImageProps {
   nftId: NFT['id']
   size?: DimensionValue
+  play?: boolean
 }
 
-const NFTImage = ({ nftId, size = '100%' }: NFTImageProps) => {
+enum NFTDataTypes {
+  image = 'image',
+  video = 'video',
+  audio = 'audio',
+  other = 'other'
+}
+
+type NFTDataType = keyof typeof NFTDataTypes
+
+const NFTImage = memo(({ nftId, size = '100%', play }: NFTImageProps) => {
   const nft = useAppSelector((s) => selectNFTById(s, nftId))
   const theme = useTheme()
 
   const [hasError, setHasError] = useState(false)
+  const [contentType, setContentType] = useState<NFTDataType>()
+
+  useEffect(() => {
+    if (nft?.image)
+      fetch(nft.image).then((res) => {
+        const contentType = res.headers.get('content-type') || ''
+        const contentTypeCategory = contentType.split('/')[0]
+
+        setContentType(contentTypeCategory in NFTDataTypes ? (contentTypeCategory as NFTDataType) : 'other')
+      })
+  }, [nft?.image])
 
   if (!nft) return null
 
   const isDataUri = nft.image.startsWith('data:image/')
 
-  return hasError ? (
-    <NFTPlaceholder size={size} />
-  ) : isDataUri ? (
-    <NFTWebView imageUri={nft.image} size={size} />
-  ) : (
+  if (!contentType) return <NFTPlaceholder size={size} showCameraOff={false} />
+
+  // Loading many videos at once is too heavy on Android.
+  // Using the image component for NFT lists on Android is sufficient.
+  if (contentType === 'video' && (Platform.OS === 'ios' || play))
+    return <NftVideo size={size} videoSource={nft.image} play={play} />
```

Comment:
Let's overlay a subtle "play" icon on video NFT thumbnails, like we do on the explorer? WDYT?